### PR TITLE
Adding crud for job posting and maintenance

### PR DESCRIPTION
**Pull request by eatskolnikov**
#WIP#

## What's new? ##

- Adding crud for job posting and maintenance

### Visual Aids ###

To be included later

### CODE REVIEW

File: AppServices/Services/JobsService.cs [hunk 1]
```diff
@@ -35,7 +36,13 @@ protected override TaskResult<Job> ValidateOnUpdate(Job entity)
         }
         public List<Job> GetByUser(int userId)
         {
-            return _mainRepository.Get(x=>x.UserId == userId).OrderByDescending(x=>x.PublishedDate).ToList();
+            return _mainRepository
+                .Get(x=>x.UserId == userId)
```
Comment (DannyFeliz) [inline AppServices/Services/JobsService.cs]:
```suggestion
                .Get(x => x.UserId == userId)
```

File: AppServices/Services/JobsService.cs [hunk 2]
```diff
@@ -35,7 +36,13 @@ protected override TaskResult<Job> ValidateOnUpdate(Job entity)
...
+                .Include(x => x.Company)
+                .Include(x => x.Category)
+                .Include(x => x.HireType)
+                .Include(x => x.Location)
+                .OrderByDescending(x=>x.PublishedDate).ToList();
```
Comment (DannyFeliz) [inline AppServices/Services/JobsService.cs]:
```suggestion
                .OrderByDescending(x => x.PublishedDate).ToList();
```

File: AppServices/Services/JobsService.cs [hunk 3]
```diff
@@ -53,13 +60,31 @@ public override List<Job> GetAll()
 
         public IEnumerable<Job> GetRecentJobs()
         {
-            return _mainRepository.Get(x=>x.IsActive && !x.IsHidden, "Company").OrderByDescending(x => x.PublishedDate).Take(10).ToList();
+            return _mainRepository.Get(x=>x.IsActive && !x.IsHidden)
```
Comment (DannyFeliz) [inline AppServices/Services/JobsService.cs]:
```suggestion
            return _mainRepository.Get(x => x.IsActive && !x.IsHidden)
```

File: Web/Controllers/JobsController.cs [hunk 4]
```diff
@@ -43,28 +48,145 @@ public async Task<IActionResult> Index(string keyword = "", bool isRemote = fals
...
                 Categories = _categoriesService.GetAll(),
-                JobTypes = _hiretypesService.GetAll()
+                JobTypes = _hiretypesService.GetAll(),
+                Companies  = _companiesService.GetByUserId(_currentUser.UserId)
             };
 
+            if (id.HasValue)
+            {
+                var originalJob = _jobsService.GetById(id.Value);
+                if(originalJob.UserId == _currentUser.UserId)
```
Comment (DannyFeliz) [inline Web/Controllers/JobsController.cs]:
```suggestion
                if (originalJob.UserId == _currentUser.UserId)
```

File: Web/Controllers/JobsController.cs [hunk 5]
```diff
@@ -43,28 +48,145 @@ public async Task<IActionResult> Index(string keyword = "", bool isRemote = fals
...
+                        };
+
+                        _companiesService.Create(company);
+                        companyId = company.Id;
+                    }
+
+                    if (model.Id.HasValue)
+                    {
+                        var originalJob = _jobsService.GetById(model.Id.Value);
+                        if(originalJob.UserId == _currentUser.UserId)
```
Comment (DannyFeliz) [inline Web/Controllers/JobsController.cs]:
```suggestion
                        if (originalJob.UserId == _currentUser.UserId)
```

File: Web/Controllers/JobsController.cs [hunk 6]
```diff
@@ -43,28 +48,145 @@ public async Task<IActionResult> Index(string keyword = "", bool isRemote = fals
...
+                        {
+
+                            originalJob.CategoryId = model.CategoryId;
+                            originalJob.HireTypeId = model.JobTypeId;
+                            originalJob.CompanyId = companyId.Value;
+                            originalJob.HowToApply = model.HowToApply;
+                            originalJob.Description = model.Description;
+                            originalJob.Title = model.Title;
+                            originalJob.IsRemote = model.IsRemote;
+                            if(originalJob.Location.PlaceId != model.LocationPlaceId)
```
Comment (DannyFeliz) [inline Web/Controllers/JobsController.cs]:
```suggestion
                            if (originalJob.Location.PlaceId != model.LocationPlaceId)
```

File: Web/Controllers/JobsController.cs [hunk 7]
```diff
@@ -43,28 +48,145 @@ public async Task<IActionResult> Index(string keyword = "", bool isRemote = fals
...
+                                originalJob.Location = new Location
+                                {
+                                    PlaceId = model.LocationPlaceId,
+                                    Name = model.LocationName,
+                                    Longitude = model.LocationLongitude,
+                                    Latitude = model.LocationLatitude
+                                };
+                            }
+                            var result = _jobsService.Update(originalJob);
+                            if(result.Success)
```
Comment (DannyFeliz) [inline Web/Controllers/JobsController.cs]:
```suggestion
                            if (result.Success)
```

File: Web/Legacy/LegacyApiClient.cs [hunk 8]
```diff
@@ -30,8 +30,8 @@ public LegacyApiClient(IFeatureManager featureManager, IConfiguration configurat
     
     public async Task<IList<JobCardDTO>> GetJobsFromLegacy()
     {
-        if(_featureManager.IsEnabled(FeatureFlags.LegacyClient.UseMockData))
-        return GetJobsFromMockData();
+        //if(_featureManager.IsEnabled(FeatureFlags.LegacyClient.UseMockData))
+           return GetJobsFromMockData();
```
Comment (DannyFeliz) [inline Web/Legacy/LegacyApiClient.cs]:
```suggestion
       return GetJobsFromMockData();
```

File: Web/Legacy/LegacyApiClient.cs [hunk 9]
```diff
@@ -30,8 +30,8 @@ public LegacyApiClient(IFeatureManager featureManager, IConfiguration configurat
     
     public async Task<IList<JobCardDTO>> GetJobsFromLegacy()
     {
-        if(_featureManager.IsEnabled(FeatureFlags.LegacyClient.UseMockData))
-        return GetJobsFromMockData();
+        //if(_featureManager.IsEnabled(FeatureFlags.LegacyClient.UseMockData))
```
Comment (DannyFeliz) [inline Web/Legacy/LegacyApiClient.cs]: Do we need this?
```suggestion
        //if(_featureManager.IsEnabled(FeatureFlags.LegacyClient.UseMockData))
```

File: Web/wwwroot/css/styles.css [hunk 10]
```diff
@@ -6903,7 +6903,7 @@ span.current-page:before {
 	background-size:cover;
 	position:relative;
 	padding:0;
-	min-height:380px;
+	1min-height:380px;
```
Comment (DannyFeliz) [inline Web/wwwroot/css/styles.css]: This is no valid CSS.

File: Web/wwwroot/css/styles.css [hunk 11]
```diff
@@ -6993,7 +6993,7 @@ span.current-page:before {
     margin-left: 7px;
 }
 .detail-swap {
-    height: 380px;
+    !height: 380px;
```
Comment (DannyFeliz) [inline Web/wwwroot/css/styles.css]: Same here

File: AppServices/Services/JobsService.cs [hunk 12]
```diff
@@ -35,12 +36,18 @@ protected override TaskResult<Job> ValidateOnUpdate(Job entity)
...
+                .Include(x => x.Category)
+                .Include(x => x.HireType)
+                .Include(x => x.Location)
+                .OrderByDescending(x=>x.PublishedDate).ToList();
         }
 
         public Job GetDetails(int id, bool isPreview = false)
         {
-            var job = _mainRepository.Get(j => j.Id == id && j.Approved == !isPreview, "Company")
+            var job = _mainRepository.Get(j => j.IsActive && j.Id == id, "Company,Location")
```
Comment (ljcordero) [inline AppServices/Services/JobsService.cs]: isPreview parameter is not being used.

Comment (ljcordero) [reply AppServices/Services/JobsService.cs]: Also what about use Include method instead of string include.

**Pull request merged**